src/lib/ngsi/ContextElement.cpp

```diff
@@ -111,6 +111,32 @@ std::string ContextElement::render(ConnectionInfo* ciP, RequestType requestType,
 
 /* ****************************************************************************
 *
+* ContextElement::toJson - 
+*/
+std::string ContextElement::toJson(ConnectionInfo* ciP, NotificationFormat notifyFormat)
```

Comment:
I'm in favour of each function having *no more and no less* of the input it need to do if work. This is the basic principle of modularity in programming lanaugages. If we provide ciP in a function that doesn't need, we are breaking the *no more* part (specially when ciP is a complex object that explodes in a lot of sub-field, making more complex).

Regarding manteinability, it depends. If there is a overall cohesion in all the methods using the same signature, it may have sense. But, in this case, I think is more manteinable (as it is easier to understand, specially for newcomers to the Orion code) `toJson(void)` than `toJson(ciP)`. If I were a developer looking the code for the first time and I see `toJson(ciP)` I will start wondering why that funciton have an argument that it doesn't use and I could end getting lost in the details of ConnectionInfo.h in order to figure out what it is all that about.

So, at least for the toJson() family, I will try to adhere to the *no more and no less* principle. Let's see how manteinable is the code and the feedback of new developers after a while of applying it.